frontend/packages/operator-lifecycle-manager/src/components/create-operand.tsx

```diff
@@ -157,20 +319,164 @@ const fieldsForOpenAPI = (openAPI: SwaggerDefinition): OperandField[] => {
+                (itemVal, itemKey: string): OperandField => ({
+                  path: [key, itemKey].join('.'),
+                  displayName: _.startCase(itemKey),
+                  type: itemVal.type,
+                  required: _.get(val.items, 'required', []).includes(itemKey),
+                  validation: null,
+                  capabilities: [
+                    SpecCapability.fieldGroup.concat(key) as SpecCapability.fieldGroup,
+                    ...capabilitiesFor(itemVal),
+                  ],
+                }),
+              );
+            }
+            if (
+              val.items.type !== 'object' ||
+              _.values(val.items.properties).every((nestedVal) =>
+                _.values(nestedVal.properties).every(
+                  (thirdLevelVal) => !['object', 'array'].includes(thirdLevelVal.type),
+                ),
+              )
+            ) {
+              return _.map(
+                val.properties,
+                (nestedVal, nestedKey: string): OperandField => {
+                  if (!['object', 'array'].includes(nestedVal.type)) {
+                    return {
+                      path: [key, nestedKey].join('.'),
+                      displayName: _.startCase(nestedKey),
+                      type: nestedVal.type,
+                      required: _.get(val, 'required', []).includes(nestedKey),
+                      validation: null,
+                      capabilities: [
+                        SpecCapability.internalFieldGroupThirdSecLevel.concat(
+                          key,
+                        ) as SpecCapability.fieldGroup,
+                        ...capabilitiesFor(nestedVal.type),
+                      ],
+                    };
+                  }
+                  if (!_.has(nestedVal, 'properties')) {
+                    return null;
+                  }
+
+                  return _.map(
+                    nestedVal.properties,
+                    (thirdLevelVal, thirdLevelKey: string): OperandField => ({
+                      path: [key, nestedKey, thirdLevelKey].join('.'),
+                      displayName: _.startCase(thirdLevelKey),
+                      type: ThirdLevelVal.type,
+                      required: _.get(val.properties, 'required', []).includes(thirdLevelKey),
+                      validation: null,
+                      capabilities: [
+                        SpecCapability.internalFieldGroupThirdLevel.concat(
+                          key,
+                        ) as SpecCapability.fieldGroup,
+                        SpecCapability.internalFieldGroupThirdSecLevel.concat(
+                          nestedKey,
+                        ) as SpecCapability.fieldGroup,
+                        ...capabilitiesFor(thirdLevelVal.type),
+                      ],
+                    }),
+                  );
+                },
+              );
+            }
+
+            if (
+              val.items.type !== 'object' ||
```

Comment:
Just pointing out again I bet a shared helper func would make this easier to manage.